feeds/pypi/pypi.go

```diff
@@ -167,19 +170,30 @@ func New(feedOptions feeds.FeedOptions, eventHandler *events.Handler) (*Feed, er
 func (feed Feed) Latest(cutoff time.Time) ([]*feeds.Package, error) {
 	pkgs := []*feeds.Package{}
 	var pypiPackages []*Package
+	var failedPackages []feeds.FailedPackagePoll
 	var err error
 
 	if feed.packages == nil {
 		// Firehose fetch all packages.
+		// If this fails then we need to return, as it's the only source of
+		// data.
 		pypiPackages, err = fetchPackages(feed.baseURL)
+		if err != nil {
+			return nil, err
+		}
 	} else {
 		// Fetch specific packages individually from configured packages list.
-		pypiPackages, err = fetchCriticalPackages(feed.baseURL, *feed.packages)
+		pypiPackages, failedPackages = fetchCriticalPackages(feed.baseURL, *feed.packages)
+		for _, pkg := range failedPackages {
+			// This could be an entry point for a 'lossy' type of event.
+			log.Errorf("Failed to poll for %s package due to err: %v", pkg.Name, pkg.Err)
+		}
+		if len(pypiPackages) == 0 {
+			// If none of the packages were successfully polled for, return early.
```

Comment:
I'm not sure returning early is necessary as the following code blocks don't do anything if there are no pypi packages. Additionally that means we don't need the `errors.Is` check.

Reply:
The `errors.Is` check was more for testing purposes to ensure that the correct error was propagated to the user. If we don't think that is necessary then I agree the early return isn't needed.